openfe_gromacs/protocols/gromacs_md/md_methods.py

```diff
@@ -0,0 +1,649 @@
+                has_solvent=solvent_comp is not None,
+            )
+
+            # Force creation of smc templates so we can solvate later
+            for mol in smc_components.values():
+                system_generator.create_system(
+                    mol.to_topology().to_openmm(), molecules=[mol]
+                )
+
+            # d. get OpenMM Modeller + a resids dictionary for each component
+            stateA_modeller, comp_resids = system_creation.get_omm_modeller(
+                protein_comp=protein_comp,
+                solvent_comp=solvent_comp,
+                small_mols=smc_components,
+                omm_forcefield=system_generator.forcefield,
+                solvent_settings=solvation_settings,
+            )
+
+            # e. get topology & positions
+            # Note: roundtrip positions to remove vec3 issues
+            stateA_topology = stateA_modeller.getTopology()
+            stateA_positions = to_openmm(from_openmm(stateA_modeller.getPositions()))
+
+            # f. create the stateA System
+            stateA_system = system_generator.create_system(
+                stateA_topology,
+                molecules=[s.to_openff() for s in small_mols],
+            )
+
+            # 3. Create the Interchange object
+            barostat_idx, barostat = forces.find_forces(
+                stateA_system, ".*Barostat.*", only_one=True
+            )
+            stateA_system.removeForce(barostat_idx)
+
+            water = OFFMolecule.from_mapped_smiles("[O:1]([H:2])[H:3]")
+            cl = OFFMolecule.from_smiles("[Cl-]", name="Cl")
+            na = OFFMolecule.from_smiles("[Na+]", name="Na")
+
+            # For now we will be using the OpenMM Topology, not OpenFF
+            # unique_molecules = [water, cl, na]
+            #
+            # for mol in smc_components.values():
+            #     unique_molecules.append(mol)
+            # topology = Topology.from_openmm(
+            #     stateA_topology, unique_molecules=unique_molecules
+            # )
+
+            stateA_interchange = Interchange.from_openmm(
+                topology=stateA_topology,
+                system=stateA_system,
+                positions=stateA_positions,
+            )
+
+            for molecule_index, molecule in enumerate(
+                stateA_interchange.topology.molecules
+            ):
+                for atom in molecule.atoms:
+                    atom.metadata["residue_number"] = molecule_index + 1
+                if len([*smc_components.values()]) > 0:
+                    if molecule.n_atoms == [*smc_components.values()][0].n_atoms:
+                        for atom in molecule.atoms:
+                            atom.metadata["residue_name"] = "UNK"
+                # molecules don't know their residue metadata, so need to
+                # set on each atom
+                # https://github.com/openforcefield/openff-toolkit/issues/1554
+                elif molecule.is_isomorphic_with(water):
```

Comment:
It's odd that this would be needed - is Interchange not loading in metadata from the openmm Topology?